lib/swoosh/adapters/smtp/helpers.ex

```diff
@@ -41,18 +41,26 @@ defmodule Swoosh.Adapters.SMTP.Helpers do
   @doc false
   def prepare_message(email, config) do
     email
-    |> prepare_headers()
+    |> prepare_headers(config)
     |> prepare_parts(email, config)
   end
 
-  defp prepare_headers(email) do
+  defp prepare_headers(email, config) do
     []
     |> prepare_additional_headers(email)
     |> prepare_mime_version()
     |> prepare_reply_to(email)
     |> prepare_subject(email)
     |> prepare_cc(email)
-    # bcc is deliberately omitted: https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.3
+    |> then(fn headers ->
+      # bcc is deliberately omitted unless explicitly asked to keep, for service providers
+      # https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.3
+      if config[:keep_bcc] do
```

Comment:
I think it might be worth pulling this `body` function out of the SMTP Adapter, maybe into the Adapter module, or creating Adapter.Helpers, WDYT?

I can possibly look into it if you'd like to apply this fix for now, but it probably won't be until next week or later.

Reply:
It's in the helper module, not in the adapter module. Not to say that it's good, I dislike the helper module, but it's not as bad as being in the adapter directly.